lms/config.py

```diff
@@ -0,0 +1,121 @@
+"""Configuration for the Pyramid application."""
+
+import os
+from dataclasses import dataclass
+from typing import Callable
+
+from pyramid.config import Configurator, aslist
+from pyramid.settings import asbool
+
+
+class SettingError(Exception):
+    pass
+
+
+def _append_trailing_slash(string):
+    """Append a slash to a string if none was there before."""
+
+    return string if string.endswith("/") else string + "/"
+
+
+def _aes_to_16_chars(string):
+    """Get 16 ascii bytes from the provided string."""
+
+    try:
+        return string.encode("ascii")[0:16] if string else None
+    except UnicodeEncodeError as err:
+        raise SettingError("LMS_SECRET must contain only ASCII characters") from err
+
+
+@dataclass(frozen=True)
+class _Setting:
+    """The properties of a setting and how to read it."""
+
+    name: str
+    read_from: str = None
+    mapping: Callable = None
```

Comment:
Not sure about this name (`mapping`), but I can't come up with a better one to suggest.

Reply:
`value_mapper`?